### PR TITLE
[PM-9317] fix duo subscriptions and org vs individual duo setup

### PR DESCRIPTION
**Pull request by jlf0dev**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-9317
## 📔 Objective

We weren't sending the organization id or the correct two factor type to the dialog when configuring Organization Duo. This fixes that and adds some better handling of the dialog callbacks.
## 📸 Screenshots

https://github.com/bitwarden/clients/assets/24985544/773d55a1-fee2-4862-8485-1ddcb63a1d1b

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/admin-console/organizations/settings/two-factor-setup.component.ts [hunk 1]
```diff
@@ -75,9 +79,18 @@ export class TwoFactorSetupComponent extends BaseTwoFactorSetupComponent {
...
-        const enabled: boolean = await lastValueFrom(duoComp.closed);
-        this.updateStatus(enabled, TwoFactorProviderType.Duo);
+        const duoComp: DialogRef<boolean, any> = TwoFactorDuoComponent.open(this.dialogService, {
+          data: {
+            authResponse: result,
+            organizationId: this.organizationId,
+          },
+        });
+        this.twoFactorSetupSubscription = duoComp.componentInstance.onChangeStatus
+          .pipe(first(), takeUntil(this.destroy$))
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/admin-console/organizations/settings/two-factor-setup.component.ts:89]: ❓ why have a `takeUntil` when there is a `first`?  [First](https://rxjs.dev/api/operators/first) will complete after the first emission.

Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/admin-console/organizations/settings/two-factor-setup.component.ts:89]: There are several other places where this comment applies as well.

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]

File: apps/web/src/app/admin-console/organizations/settings/two-factor-setup.component.ts (hunk 1 above)
Comment (JaredSnider-Bitwarden) [reply apps/web/src/app/admin-console/organizations/settings/two-factor-setup.component.ts:89]: Nvm. Per Jake: " if the dialog is closed and component left the subscription will be kept alive. Basically, if there’s no emission I wanted it to still be destroyed with the component"

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by eliykat — APPROVED** [review]
I think it would be preferable for the 2FA dialogs to call `dialogRef.close` with a result so that the `TwoFactorSetupComponent` can just `await lastValueFrom(dialogRef.closed)`. That way you don't have to define your own `EventEmitter` and manage your subscription etc. However that is non-blocking feedback.

**Pull request merged**